pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1029,10 +1029,28 @@ func listAttestedNodes(ctx context.Context, db *sqlDB, req *datastore.ListAttest
 
 		resp.Nodes = filterNodesBySelectorSet(resp.Nodes, req.BySelectorMatch.Selectors)
 
+		// Now that we've filtered the nodes based on selectors, prune off
+		// selectors from the response if they were not requested.
+		if !req.FetchSelectors {
```

Comment:
Having just implemented this in the KV prototype, this is a change in implementation. Is/was this documented? Was it incorrect or are we changing it?

Reply:
The behavior isn't correct, or externally documented. The datastore should only return the selectors if requested, however this bug is not externally noticeable since:
1 - the agent API asks for selectors by default unless masked. Even if the datastore returns them when it shouldn't, the `output_mask` will cause the selectors to be dropped before returning the response.
2 - the registration API never asks filters by selector so the bug is never activated.